bundles/org.eclipse.jface/src/org/eclipse/jface/operation/ModalContext.java

```diff
@@ -255,7 +258,9 @@ public static boolean canProgressMonitorBeUsed(IProgressMonitor monitor1, IProgr
 	 */
 	public static void checkCanceled(IProgressMonitor monitor) throws InterruptedException {
 		if (monitor.isCanceled()) {
-			throw new InterruptedException();
+			InterruptedException interruptedException = new InterruptedException();
+			interruptedException.initCause(new OperationCanceledException());
+			throw interruptedException;
```

Comment:
This is semantically incorrect. A monitor may be cancelled for other reasons than the one stated in an `OperationCanceledException`, which is tied to the user canceling an operation. In addition, there is no information gain in adding another exception as a cause to the `InterruptedException`, as no further data is added to the exception and the stack trace will be the same. So it only add overhead due to exception generation.